modules/healthconnectonfhir/src/main/java/edu/stanford/healthconnectonfhir/HealthConnectToFHIR.kt

```diff
@@ -0,0 +1,195 @@
+package edu.stanford.healthconnectonfhir
+
+import androidx.health.connect.client.records.ActiveCaloriesBurnedRecord
+import androidx.health.connect.client.records.BloodPressureRecord
+import androidx.health.connect.client.records.BodyTemperatureRecord
+import androidx.health.connect.client.records.HeartRateRecord
+import androidx.health.connect.client.records.HeightRecord
+import androidx.health.connect.client.records.Record
+import androidx.health.connect.client.records.StepsRecord
+import androidx.health.connect.client.records.WeightRecord
+import org.hl7.fhir.r4.model.CodeableConcept
+import org.hl7.fhir.r4.model.Coding
+import org.hl7.fhir.r4.model.Observation
+import org.hl7.fhir.r4.model.Period
+import org.hl7.fhir.r4.model.Quantity
+import java.util.Date
+
+private fun <T: Record> T.createObservation(
```

Comment:
Thank you for this detailed feedback @eldcn! It makes sense to me. One of the reasons we are using extensions here is for this package to resemble the API surface of its corresponding iOS package - https://github.com/StanfordBDHG/HealthKitOnFHIR. However, as you mention, creating a custom object has its benefits. It would be great to have @PSchmiedmayer and @Basler182's thoughts on this also.

Reply:
I intentionally haven't checked the iOS project in detail to not get biased from iOS specific architecture 😅 

Let me bring you an (hypothetical, or maybe not so) example. I think there is no better measurement for a software system, then how easy or hard it is to introduce a change into it.

Let's suppose an application using these extension functions supports localization -> we might need to extract the strings into resources and provide different variants for the display text. In order to read the resources, we need an application `context`.

Now let's evaluate both options considering that we are using these extension functions in multiple places in the project.

1. Extension functions: We would need to update all the usages and provide a context parameter in every `toObservation` method. If a context isn't available, we would again need to inject it in the component using the extension function, and also update it's tests.

2. Mapper: You just inject the context in the mapper constructor, and update it's test only. All `mapper.map(record)` usages remain untouched

Reply:
This rationale makes a lot of sense and I would agree with it. I want to balance it also with the desire to keep the API surface similar between the iOS and Android frameworks, so that developers can more easily port their code from one to the other. This is something developers using CardinalKit/Spezi have always asked for. Would be great to hear from others as well.

Reply:
@vishnuravi , I see advantages in @eldcn suggestion. With other modules, I have mostly worked with interfaces and used DI mechanisms. Although I have no experience with iOS, and the reasons why it has be done in other ways. I think it can be benefical to continue to develop the framework and introduce new concepts. We will make them documented and easy to understand, which is why I think we should pursue this approach. But that's just my personal opinion. I'm curious what @PSchmiedmayer  has to say about it

Reply:
Thanks @eldcn and @Basler182 for your thoughts on this. WIll let @PSchmiedmayer weigh in also.

Reply:
I agree with your reasoning here @PSchmiedmayer. We can discuss further and come to a decision at the team meeting today.